sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/NotificationHook/NotificationHook.cs

```diff
@@ -64,12 +64,33 @@ internal NotificationHook(HookType hookType, string id, string name, string desc
 
         internal static HookInfoPatch GetPatchModel(NotificationHook hook)
         {
-            return hook switch
+            HookInfoPatch patch = hook switch
```

Comment:
Same as in other PRs. The service uses a different type for Update, so we need to convert here.